# Rough Work for Individual Submission (all relevant things copied to the main document)



In [5]:
#attach packages to work with data
library(tidyverse)
library(repr)
library(tidymodels)
options(repr.matrix.max.rows = 6)
#source("cleanup.R")



## opening files, making them tidy


In [11]:
#open files for descriptive summaries
players_data = read_csv("players.csv")
sessions_data = read_csv("sessions.csv")

players_data
sessions_data



Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1   Pro         TRUE    
2   Veteran     TRUE    
3   Veteran    FALSE    
⋮   ⋮          ⋮        
194 Amateur    FALSE    
195 Amateur    FALSE    
196 Pro         TRUE    
    hashedEmail                                                     
1   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2   f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
3   b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28
⋮   ⋮                                                               
194 d572f391d452b76ea2d7e5e53a3d38bfd7499c7399db299bd4fedb06a46ad5bb
195 f19e136ddde68f365afc860c725ccff54307dedd13968e896a9f890c40aea436
196 d9473710057f7d42f36570f0be83817a4eea614029ff90cf50d8889cdd729d11
    played_hours name      gender            Age
1   30.3         Morgan    Male               9 
2    3.8         Christian Male              17 
3    0.0         Blake     Male              17 
⋮   ⋮            ⋮         ⋮                 ⋮  
194 0.0          Dylan     Prefer not to say 57 
195 2.3          Harlow    Male              17 
196 0.2          Ahmed     Other             NA

hashedEmail                                                     
1    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3    f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
⋮    ⋮                                                               
1533 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1534 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1535 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
     start_time       end_time         original_start_time original_end_time
1    30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2    17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3    25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
⋮    ⋮                ⋮                ⋮                   ⋮                
1533 28/07/2024 15:36 28/07/2024 15:57 1.72218e+12         1.72218e+12      
1534 25/07/2024 06:15 25/07/2024 06:22 1.72189e+12         1.72189e+12      
1535 20/05/2024 02:26 20/05/2024 02:45 1.71617e+12         1.71617e+12

In [13]:
sessions_wrangled = sessions_data |>
    separate(start_time, into = c("start_date", "start_time"), sep = " ") |>
    mutate(start_date = as.Date(start_date)) |>
    separate(end_time, into = c("end_date", "end_time"), sep = " ") |>
    mutate(end_date = as.Date(end_date))

sessions_wrangled


hashedEmail                                                     
1    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3    f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
⋮    ⋮                                                               
1533 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1534 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1535 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
     start_date start_time end_date end_time original_start_time
1    30-06-20   18:12      30-06-20 18:24    1.71977e+12        
2    17-06-20   23:33      17-06-20 23:46    1.71867e+12        
3    25-07-20   17:34      25-07-20 17:57    1.72193e+12        
⋮    ⋮          ⋮          ⋮        ⋮        ⋮                  
1533 28-07-20   15:36      28-07-20 15:57    1.72218e+12        
1534 25-07-20   06:15      25-07-20 06:22    1.72189e+12        
1535 20-05-20   02:26      20-05-20 02:45    1.71617e+12        
     original_end_time
1    1.71977e+12      
2    1.71867e+12      
3    1.72193e+12      
⋮    ⋮                
1533 1.72218e+12      
1534 1.72189e+12      
1535 1.71617e+12

## Getting descriptive summaries of datasets


In [18]:
#players_data
players_summary = players_data |>
    summary()
players_summary

experience_types = unique(players_data$experience) #$ access unique values from experience in players_data
experience_types

gender_types = unique(players_data$gender)
gender_types


  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

[1] "Pro"      "Veteran"  "Amateur"  "Regular"  "Beginner"

[1] "Male"              "Female"            "Non-binary"       
[4] "Prefer not to say" "Agender"           "Two-Spirited"     
[7] "Other"

### Dataset Summary: players_data

The `players_data` dataset contains 196 observations (i.e., 196 players). There are seven different variables. Below are the variables' names, types, and meanings.

- `experience`: character variable, 5 categories ("Pro", "Veteran", "Amateur", "Regular", "Beginner") based on how much experience the player has with Minecraft.
- `subscribe`: logical variable, returns "TRUE" or "FALSE" depending if the player is currently subscribed to a game-related newsletter
- `hashedEmail`: character variable returning the player's email, irreversibly transformed to a unique string (facilitating individualization while retaining privacy).
- `played_hours`: double-class variable corresponding to how many hours the player has spent playing the game.
- `name`: character variable containing the player's name (likely first name).
- `gender`: character variable, 7 categories ("Male", "Female", "Non-binary", "Prefer not to say", "Agender", "Two-Spirited", "Other"), self-reported by the player.
- `Age`: double-class variable returning the player's age in years (self-reported).

While summary statistics can be individually computed with commands like `mean()`, `min()`, or `max()`, in this case it is efficient to collect them all at once with `summary()`. There are three different variables in `players_data` for which summary statistics can be calculated (barring general statistics like `length`, `class`, and `mode`). Summary statistics have been rounded to two decimal places.

| Variable | min | 1st quartile | median | mean | 3rd quartile | max | NA's |
| -------- | --- | ------------ | ------ | ---- | ------------ | --- | ---- |
| `played_hours` (hours) | 0.00 | 0.00 | 0.10 | 5.85 | 0.60 | 223.10 | -- |
| `age` (years) | 9.00 | 17.00 | 19.00 | 21.14 | 22.75 | 58.00 | 2 |

For `age`, NA's means that two players opted not to share their ages.
The third variable in `players_data` with summary statistics is `subscribe`, as such:

| Variable | TRUE | FALSE |
| -------- | ---- | ----- |
| `subscribe` | 144 | 52 |

This just means that 144 players in the dataset are subscribed to a game-related newsletter.


  
  